### PR TITLE
docs: Add data plane signaling documents

### LINKED ISSUE #3704: docs: Add an architecture document and decision record for Data Plane Signaling
Adds architecture documentation and a decision record covering Data Plane Signaling.

## Which Areas Would Be Affected?

Core, DPF

### PR DESCRIPTION
**Pull request by jimmarino**
## What this PR changes/adds

Adds architecture documentation and a decision record covering Data Plane Signaling.

## Linked Issue(s)

Closes #3704

### CODE REVIEW

**Review by paullatzelsperger — APPROVED** [review]

File: docs/developer/decision-records/2023-12-12-dataplane-signaling/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+Data Plane Signaling and Access Control Architecture
```
Comment (paullatzelsperger) [inline docs/developer/decision-records/2023-12-12-dataplane-signaling/README.md:1]: could you link this decision record in `docs/developer/decision-records/README.md`?

Comment (paullatzelsperger) [inline docs/developer/decision-records/2023-12-12-dataplane-signaling/README.md:1]:
```suggestion
# Data Plane Signaling and Access Control Architecture
```

File: docs/developer/decision-records/2023-12-12-dataplane-signaling/README.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+# Decision
```
Comment (paullatzelsperger) [inline docs/developer/decision-records/2023-12-12-dataplane-signaling/README.md:3]:
```suggestion
## Decision
```

File: docs/developer/decision-records/2023-12-12-dataplane-signaling/README.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+EDC will implement the Data Plane Signaling Protocol as described [here](../../data-plane-signaling/data-plane-signaling.md). The Signaling Protocol defines how a control plane starts, suspends, and terminates a data flow in a data plane.
+
+# Rationale
```
Comment (paullatzelsperger) [inline docs/developer/decision-records/2023-12-12-dataplane-signaling/README.md:7]:
```suggestion
## Rationale
```

File: docs/developer/decision-records/2023-12-12-dataplane-signaling/README.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+The EDC control plane currently implements an unspecified communication protocol when initiating flows in its proprietary data plane implementation, the DPF. In addition to being proprietary and undocumented, the communication has significant limitations. These include the inability to suspend flows and lack of adequate access token management.  
+
+The Data Plane Signaling Protocol will be the foundation for the EDC control plane to interoperate with third-party data planes. It defines standardized messages and access control behavior that allow third-party data planes to work out-of-the-box with EDC-based control planes.
+
+# Impact
```
Comment (paullatzelsperger) [inline docs/developer/decision-records/2023-12-12-dataplane-signaling/README.md:13]:
```suggestion
## Impact
```

File: docs/developer/data-plane-signaling/data-plane-signaling.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+### The Signaling Protocol
+
+All requests must support idempotent behavior. Data planes must therefore perform request de-duplication. After a data plane commits a request, it will return an ack to the control plane, which will transition the `TransferProcess` to its next state (e.g., `STARTED`, `SUSPENDED`, `TERMINATED`). If a successful ack is not received, the control plane will resend the request during a subsequent tick period.
+
+##### 1. `START` 
+
+During the transfer process `STARTING` phase, a data plane will be selected by the default push and pull `DataFlowControllers`, which will then send a `DataFlowStartMessage` (renamed `DataFlowRequest`) to the data plane. 
+
+The control plane (`DataFlowController`) will record which data plane was selected for the transfer process so that it can properly route subsequent, start, stop, and terminate requests.
```
[the remark this answers is not in the document]
Comment (jimmarino) [reply docs/developer/data-plane-signaling/data-plane-signaling.md:48]: Each data plane can be completely independent and will likely not share the same persistent storage. The control plan will need to track which data plane the request has been assigned to.

**Review by wolf4ood — APPROVED** [review]

**Review by ronjaquensel — APPROVED** [review]

**Review by ndr-brt — APPROVED** [review]

**Pull request merged**